src/path.jl

```diff
@@ -765,3 +765,21 @@ Base.mkpath(fp::AbstractPath) = mkdir(fp; recursive=true)
 # ALIASES for now old FilePaths API
 move(src::AbstractPath, dest::AbstractPath; kwargs...) = mv(src, dest; kwargs...)
 remove(fp::AbstractPath; kwargs...) = rm(fp; kwargs...)
+
+"""
+	isdescendent(child::AbstractPath, parent::AbstractPath) -> Bool
+
+Returns `true` if `child` is within the directory tree of `parent`.
+"""
+function isdescendent(child::AbstractPath, parent::AbstractPath)
+	cn = normpath(child).segments
+	pn = normpath(parent).segments
+	return length(pn) <= length(cn) && all(cn[1:length(pn)] .== pn)
+end
+
+"""
+	isascendant(parent::AbstractPath, child::AbstractPath) -> Bool
+
+Returns `true` if `parent` is a directory containing `child`.
+"""
+isascendant(parent::AbstractPath, child::AbstractPath) = isdescendent(child, parent)
```

Comment:
```suggestion
"""
	isdescendant(fp::P, asc::P) where {P <: AbstractPath} -> Bool

Returns `true` if `fp` is within the directory tree of the `asc`.
"""
function isdescendant1(child::AbstractPath, parent::AbstractPath)
	cn = normpath(child).segments
	pn = normpath(parent).segments
	return length(pn) <= length(cn) && all(cn[1:length(pn)] .== pn)
end

isdescendant2(fp::P, asc::P) where {P <: AbstractPath} = fp == asc || asc in parents(fp)

"""
	isascendant(fp::P, desc::P) where {P <: AbstractPath} -> Bool

Returns `true` if `fp` is a directory containing `desc`.
"""
isascendant(fp::P, desc::P) where {P <: AbstractPath} = isdescendant(fp, desc)
```

Alright, this seems to pass your tests and seems to have about the same performance. NOTE: I don't thing we actually want to perform `normpath` since folks can always do that themselves.

```
julia> @benchmark isdescendant1($fp, $asc)
BenchmarkTools.Trial:
  memory estimate:  1.23 KiB
  allocs estimate:  28
  --------------
  minimum time:     3.057 μs (0.00% GC)
  median time:      3.204 μs (0.00% GC)
  mean time:        3.422 μs (0.98% GC)
  maximum time:     340.644 μs (98.51% GC)
  --------------
  samples:          10000
  evals/sample:     8

julia> @benchmark isdescendant2($fp, $asc)
BenchmarkTools.Trial:
  memory estimate:  1.02 KiB
  allocs estimate:  28
  --------------
  minimum time:     3.177 μs (0.00% GC)
  median time:      3.356 μs (0.00% GC)
  mean time:        3.509 μs (0.75% GC)
  maximum time:     270.841 μs (97.60% GC)
  --------------
  samples:          10000
  evals/sample:     8
```